# Day 1 Task — Google Play Store Apps: Exploratory Data Analysis

**Dataset:** Google Play Store Apps
**Goal:** Understand the dataset, frame a business problem, choose an ML approach, and document
initial findings using Pandas.


## 1. Dataset Purpose

This dataset contains **10,841 Android apps** from the Google Play Store, described by 13 columns
covering category, rating, number of reviews, size, installs, price, content rating, genre, and
version info. It's useful for understanding what drives app popularity and how developers price
and position apps on the store.


## 2. Setup & Load Data

In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv("googleplaystore.csv")

### Preview the data

In [3]:
df.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


## 3. Business Problem(s)

1. **Identify app categories with high demand but low competition** by comparing average installs
   against the number of apps per category.
2. **Determine the optimal price range for paid apps** by analyzing the relationship between price
   and installs.
3. **Identify the key factors influencing app popularity** — category, rating, reviews, size,
   type, and pricing.


## 4. ML Problem Framing

**Type: Binary Classification.** The goal is to predict whether an app is **High Popularity** or
**Low Popularity**, with the label derived from `Installs`. Category, Rating, Reviews, Size, Type,
Price, Content Rating, Genres, and Android Version are used as input features.

**Target variable:** App Popularity (derived from `Installs`)
**Key features:** `Category`, `Rating`, `Reviews`, `Size`, `Type`, `Price`, `Content Rating`,
`Genres`, `Current Ver`, `Android Ver`


## 5. Basic Exploration with Pandas

### 5.1 Missing values

In [4]:
df.isna().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

### 5.2 Dataset shape

In [5]:
df.shape

(10841, 13)

### 5.3 Columns

In [6]:
df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='str')

### 5.4 Data types & non-null counts

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  str    
 1   Category        10841 non-null  str    
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  str    
 4   Size            10841 non-null  str    
 5   Installs        10841 non-null  str    
 6   Type            10840 non-null  str    
 7   Price           10841 non-null  str    
 8   Content Rating  10840 non-null  str    
 9   Genres          10841 non-null  str    
 10  Last Updated    10841 non-null  str    
 11  Current Ver     10833 non-null  str    
 12  Android Ver     10838 non-null  str    
dtypes: float64(1), str(12)
memory usage: 1.1 MB


### 5.5 Summary statistics

In [8]:
df.describe()

,Rating
count,9367.000000
mean,4.193338
std,0.537431
min,1.000000
25%,4.000000
50%,4.300000
75%,4.500000
max,19.000000


### 5.6 Duplicate rows

In [9]:
df.duplicated().sum()

np.int64(483)

### 5.7 Rating range sanity check

In [10]:
# Rating should be between 1 and 5 — check for out-of-range values
df[df['Rating'] > 5][['App', 'Category', 'Rating']]


,App,Category,Rating
10472,Life Made WI-Fi Touchscreen Photo Frame,1.9,19.0


## 6. Three Key Observations

1. **Missing values are concentrated in a few columns.** `Rating` is missing for 1,474 of 10,841
   apps (~14%), while `Type`, `Content Rating`, `Current Ver`, and `Android Ver` are each missing
   only a handful of rows — these need preprocessing (e.g., median imputation for `Rating`, mode
   or "Unknown" for the rest) before modeling.
2. **483 duplicate rows exist** and should be dropped — they'd otherwise bias any model trained
   on this data toward whichever apps happen to be repeated.
3. **There's a data-quality outlier worth fixing before modeling**: one row has `Rating = 19`,
   impossible on Play Store's 1–5 scale — a sign of shifted/misaligned columns in that row that
   needs to be corrected or dropped, not just capped.


## 7. Suggested Next Steps

- Drop the 483 duplicate rows and fix or drop the mis-parsed `Rating = 19` row.
- Convert `Installs`, `Price`, and `Size` from strings (e.g., `"10,000+"`, `"$4.99"`, `"19M"`) to
  numeric types.
- Define the popularity label (e.g., top-N% of installs = High) and train a baseline classifier
  (Logistic Regression or Random Forest) on the cleaned features.
